sdk/src/azure/core/az_http_response.c

```diff
@@ -158,6 +163,12 @@ AZ_NODISCARD az_result az_http_response_get_next_header(
     }
   }
 
+  if (az_span_size(ref_response->_internal.parser.remaining) == 0)
```

Comment:
A bug fix like this should be accompanied with unit tests. Please make sure to add the tests.